lib/Graphs/ICFG.cpp

```diff
@@ -460,11 +471,15 @@ struct DOTGraphTraits<ICFG*> : public DOTGraphTraits<PAG*>
         if (IntraBlockNode* bNode = SVFUtil::dyn_cast<IntraBlockNode>(node))
         {
             rawstr << "IntraBlockNode ID: " << bNode->getId() << " \t";
-            PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
-            for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
-            {
-                const PAGEdge* edge = *it;
-                rawstr << edge->toString();
+            rawstr << value2String(bNode->getInst()) << " \t";
+            PAG* pag = PAG::getPAG(false, true /* skipCreate */);
+            if (pag != nullptr) {
+                PAG::PAGEdgeList&  edges = pag->getInstPTAPAGEdgeList(bNode);
+                for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
+                {
+                    const PAGEdge* edge = *it;
+                    rawstr << edge->toString();
+                }
```

Comment:
I would suggest making an option in `Options.cpp` for users to choose whether they would like to dump the PAG information or not.

Reply:
That seems like a good idea, though I think if the PAG is available one would want that information when dumping the ICFG. But when dumping from within the debugger, you may be at a point where the PAG has not yet been created. In that case, if you had set the option to dump the PAG info, the ICFG dumper would still need to check whether the PAG was present, as you would not want to trigger creation of the PAG from withing the dumper (which is what was happening to me). I suggest not adding an option, but to avoid complicating getPAG, I'll add a different method, say, getPAGIfAvailable, that can be used by the dumper. Would that we acceptable?

Reply:
Hmm, sometimes, when debugging, we would like to correlate an ICFGNode with a PAGEdge. This is particularly useful for connecting control-dependence and data-dependence. Your way can not achieve this. 

Note that ICFG is created inside the PAG's constructor, which means PAG will always available if you try to use ICFG. It looks a bit odd trying to get a null PAG instance just for the purpose of debugging ICFG.

Reply:
You can make a change to add an option as a default one to dump only ` value2String(bNode->getInst())` but not PAGEdges. It will be a good compromise for someone who would like to see the connections between PAG and ICFG later.